sdk/messaging/azservicebus/internal/stress/tests/constant_detachment_sender.go

```diff
@@ -109,6 +109,12 @@ func checkMessages(sc *shared.StressContext, queueName string, numSent int, stat
 		defer cancel()
 
 		messages, err := receiver.ReceiveMessages(ctx, numSent, nil)
+
+		if errors.Is(err, context.DeadlineExceeded) || errors.Is(err, context.Canceled) {
+			// test is over or we received nothing (ie, we're probably done)
```

Comment:
Can we validate that we are in fact done and panic otherwise

Reply:
In this case it's a two part check - the test is done because all the messages have been drained.

And if I'm wrong there's the check after the for loop intended to validate it:

```go
if numSent != len(all) {
  sc.PanicOnError(fmt.Sprintf("Incorrect # of messages. Expected %d, got %d", numSent, len(all)), errors.New("Bad"))
}
```